### PR TITLE
Variables: Fixes maximum call stack bug for empty value

### LINKED ISSUE #25485: Variables: Text input variable does not work because "Maximum call stack size exceeded"
Text variables in dashboards does not work on master.
Screenshot from 2020-06-09 14-54-34
Screenshot from 2020-06-09 14-54-12
Screenshot from 2020-06-09 14-54-09

Seems like here https://github.com/grafana/grafana/blob/00a9af00fc8a4c6d75dedd164da545e80c79a791/public/app/features/variables/textbox/actions.ts#L28 the value from url is always empty string what ever I type in and then it fires this event in cycle.

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
When using an empty value for a Text variable the logic that checked for variable changes in URL got into an endless loop. This PR tries to remedy this.

**Which issue(s) this PR fixes**:
Fixes #25485

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hugohaggmark** [comment]
Is there some additional work that is needed before approval here @aocenas @tskarhed

**Review by tskarhed — APPROVED** [review]
Looks good! Error is no longer there!

**Comment by aocenas** [comment]
@hugohaggmark 
So there was still some issues based on the conversations in slack mainly:
> But you should not come to the state when you see `query: some text` while the actual value is something else. 

This still seems like a bug to me.

**Comment by aocenas** [comment]
But I am ok if that is solved separately.

**Comment by hugohaggmark** [comment]
Would be great if you could file that as a separate issue @aocenas 🙏

**Pull request merged**
